lib/org/openpsa/user/accounthelper.php

```diff
@@ -95,23 +95,9 @@ public function create_account(string $person_guid, string $username, string $us
         }
 
         //send welcome mail?
-        if ($send_welcome_mail) {
-            $mail = new org_openpsa_mail();
-            $mail->to = $usermail;
-
-            // Make replacements to body
-            $mail->parameters = [
-                "USERNAME" => $username,
-                "PASSWORD" => $password,
-            ];
-
-            $this->prepare_mail($mail);
-
-            if (!$mail->send()) {
-                $this->errstr = "Unable to deliver welcome mail: " . $mail->get_error_message();
-                $this->delete_account();
-                return false;
-            }
+        if ($send_welcome_mail && (!$this->send_mail($username, $usermail, $password))) {
+            $this->delete_account();
+            return false;
```

Comment:
if we unconditionally return here, we dont need `else` in the line below

Reply:
Also, it is a bit confusing: With the new version, it's possible that we go to both branches if `$send_welcome_mail` and `$generated_password` are both true. In the previous version, if `$send_welcome_mail` was true, we would never hit the `elseif` branch

Reply:
changed